torch/_torch_docs.py

```diff
@@ -3274,11 +3274,12 @@ def merge_dicts(*dicts):
            r"""
 fmod(input, other, *, out=None) -> Tensor
 
-Computes the element-wise remainder of division.
-
-The dividend and divisor may contain both for integer and floating point
-numbers. The remainder has the same sign as the dividend :attr:`input`.
+Computes the element-wise remainder of division with the remainder having the same
+sign as the dividend :attr:`input`.
 
+.. math::
+    \text{{out}}_i = \text{{input}}_i - trunc(\frac{\text{{input}}_i}{\text{{other}}_i}) * \text{{other}}_i
```

Comment:
Upon further inspection, this formula is not correct. See the output of:
```
import torch

a = torch.tensor([1.1, 1.0, 0.9, 0., -0.9, -1.0, -1.1])
for x in a:
    for y in a:
        print(f"{x}, {y}: {torch.dist(torch.remainder(x, y), x - torch.trunc(x / y)*y)}")
```

Reply:
It seems like we indeed have that this function is equivalent to `out = input - torch.trunc(input / other) * other` at least as a function from oat to float on most values (I have not tested casting nor broadcasting).

Now, when `other = float("inf")` (and same for `-float("inf")`), `fmod` returns `input`, while the expression above returns `nan`. As such, this would fall in the case where "`fmod` is a numerically stable implementation of that formula". In view of this, perhaps this could be written as:

```
[First line as it is now].

In other words, this function provides a fast and numerically stable implementation of

.. code:: python

    out = input - torch.trunc(input / other) * other
```

Reply:
Nevermind, it seems like this is not the case (https://github.com/pytorch/pytorch/pull/57941#discussion_r635318198). We need to find a good formula that shows what this operation does in terms of simpler functions and put it here.

Reply:
Thanks, @Lezcano for taking a look and also helping with the documentation. While this does make sense for `fmod` but for `remainder`, this gives accurate results:

```
# x and y are two sample torch tensors
remainder_out_formula = x - torch.round(x/y) *y
if (remainder_out_formula != 0 and ((remainder_out_formula < 0) != (y < 0))):
    remainder_out_formula += y
```

I'm confused for how to write this in words. Should it be a conditional formula, something like:

![image](https://user-images.githubusercontent.com/19997320/118834859-1fc32580-b8e0-11eb-8315-cc21d9553b53.png)

I just modified the comparison script (that you shared before, thanks for that btw):

```python
import torch

a = torch.tensor([1.1, 1.0, 0.9, 0., -0.9, -1.0, -1.1])
for x in a:
    for y in a:
        remainder_out_formula = x - torch.round(x/y) *y
        remainder_out = torch.remainder(x, y)
        if (remainder_out_formula != 0 and ((remainder_out_formula < 0) != (y < 0))):
            remainder_out_formula += y
        print(f"{x}, {y}: {torch.dist(remainder_out, remainder_out_formula)}")
        print(f"{x}, {y}: {torch.dist(torch.fmod(x, y), x - torch.trunc(x / y) *y)}")
        print('='*10)
```

Reply:
Minor update: I updated the formula, sorry, there was a typo. @Lezcano - if you can take a look again please.